test_vm/tests/extend_sectors_test.rs

```diff
@@ -302,3 +310,228 @@ fn extend_legacy_sector_with_deals_inner(do_extend2: bool) {
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::CurrentTotalPower as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::UpdatePledgeTotal as u64,
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::UpdateClaimedPower as u64,
+                params: Some(
+                    IpldBlock::serialize_cbor(&expected_update_claimed_power_params).unwrap(),
+                ),
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+
+    // inspect sector info
+
+    let miner_state = v.get_state::<MinerState>(miner_id).unwrap();
+    let sector_info_after_update = miner_state.get_sector(&store, sector_number).unwrap().unwrap();
+    assert_eq!(StoragePower::zero(), sector_info_after_update.deal_weight); // 0 space time
+
+    assert_eq!(
+        DealWeight::from((sector_info_after_update.expiration - v.get_epoch()) * (32i64 << 30)),
+        sector_info_after_update.verified_deal_weight
+    ); // 32 GiB * the remaining life of the sector
+
+    // extend the updated sector
+
+    let market_state: MarketState = v.get_state(STORAGE_MARKET_ACTOR_ADDR).unwrap();
+    let deal_states = DealMetaArray::load(&market_state.states, v.store).unwrap();
+    let deal_state = deal_states.get(deal_ids[0]).unwrap().unwrap();
+    let claim_id = deal_state.verified_claim;
+
+    let extension_params = ExtendSectorExpiration2Params {
+        extensions: vec![ExpirationExtension2 {
+            deadline: d_idx,
+            partition: partition_index,
+            sectors: BitField::new(),
+            new_expiration: sector_info_after_update.expiration + 60 * EPOCHS_IN_DAY,
+            sectors_with_claims: vec![SectorClaim {
+                sector_number,
+                maintain_claims: vec![claim_id],
+                drop_claims: vec![],
+            }],
+        }],
+    };
+    apply_ok(
+        &v,
+        worker,
+        miner_id,
+        TokenAmount::zero(),
+        MinerMethod::ExtendSectorExpiration2 as u64,
+        Some(extension_params),
+    );
+
+    let miner_state = v.get_state::<MinerState>(miner_id).unwrap();
+    let sector_info_after_extension =
+        miner_state.get_sector(&store, sector_number).unwrap().unwrap();
+    assert_eq!(StoragePower::zero(), sector_info_after_extension.deal_weight); // 0 space time
+
+    assert_eq!(
+        DealWeight::from((sector_info_after_extension.expiration - v.get_epoch()) * (32i64 << 30)),
+        sector_info_after_extension.verified_deal_weight
+    ); // 32 GiB * the remaining life of the sector
```

Comment:
please also check:

* power_base_epoch is set to current epoch
* activation is unchanged
* replaced day reward is set to the previous values day reward